rcljava/src/main/cpp/org_ros2_rcljava_action_ActionServerImpl_GoalHandleImpl.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 ros2-java contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <jni.h>
+
+#include <cassert>
+#include <cstdlib>
+#include <string>
+
+#include "rcl/error_handling.h"
+#include "rcl_action/rcl_action.h"
+
+#include "rcljava_common/exceptions.hpp"
+#include "rcljava_common/signatures.hpp"
+
+#include "org_ros2_rcljava_action_ActionServerImpl_GoalHandleImpl.h"
+
+using rcljava_common::exceptions::rcljava_throw_rclexception;
+using rcljava_common::signatures::convert_from_java_signature;
+using rcljava_common::signatures::destroy_ros_message_signature;
+
+JNIEXPORT jlong
+JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_00024GoalHandleImpl_nativeAcceptNewGoal(
```

Comment:
mangling of the function name is weird, but I guess that's what java wants here

Reply:
Yeah, it's odd, but necessary for nested classes. Alternatively, we could put the GoalHandleImpl outside of ActionServerImpl, but I like avoiding giving the user access to instantiate goal handles.